src/main/java/org/threadly/litesockets/tcp/SSLClient.java

```diff
@@ -0,0 +1,311 @@
+package org.threadly.litesockets.tcp;
+
+import static javax.net.ssl.SSLEngineResult.HandshakeStatus.FINISHED;
+import static javax.net.ssl.SSLEngineResult.HandshakeStatus.NEED_TASK;
+import static javax.net.ssl.SSLEngineResult.HandshakeStatus.NEED_UNWRAP;
+import static javax.net.ssl.SSLEngineResult.HandshakeStatus.NEED_WRAP;
+import static javax.net.ssl.SSLEngineResult.HandshakeStatus.NOT_HANDSHAKING;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.nio.channels.SocketChannel;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLEngineResult;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.threadly.litesockets.Client;
+import org.threadly.litesockets.Client.Reader;
+import org.threadly.litesockets.utils.MergedByteBuffers;
+import org.threadly.util.ExceptionUtils;
+
+
+public class SSLClient extends TCPClient implements Reader{
+  private static final String SSL_ERROR = "Problem connecting SSL Client";
+  private static final String SSL_HANDSHAKE_ERROR = "Problem doing SSL Handshake";
+  private static final TrustManager[] OPEN_TRUST_MANAGER = new TrustManager [] {new GenericTrustManager() };
+  private static final SSLContext OPEN_SSL_CTX; 
+  static {
+    //System.setProperty ("jsse.enableSNIExtension", "false");
+    try {
+      OPEN_SSL_CTX = SSLContext.getInstance("TLS");
+      OPEN_SSL_CTX.init(null, OPEN_TRUST_MANAGER, null);
+    } catch (KeyManagementException | NoSuchAlgorithmException e) {
+      throw new RuntimeException(e);
+    }
+  }
+  private final SSLEngine ssle;
+  
+  private ByteBuffer tmpAppBuffer;
+  private ByteBuffer writeBuffer;
+  
+  private final ByteBuffer encryptedReadBuffer;
+  private ByteBuffer decryptedReadBuffer;
+  
+  private volatile Reader sslReader; 
+  
+  private MergedByteBuffers decryptedReadList = new MergedByteBuffers();
+
+  public SSLClient(String host, int port) throws IOException {
+    super(host, port);
+    try {
+      long start = System.currentTimeMillis();
+      //We dont allow SSL by default connections anymore
+      ssle = OPEN_SSL_CTX.createSSLEngine(host, port);
+      ssle.setUseClientMode(true);
+      System.out.println("HS:"+ (System.currentTimeMillis()-start));
```

Comment:
I would like to avoid using System.currentTimeMillis...I would use Clock.accurateTimeMillis() as an alternative

I also would try to avoid using System.out. and System.err...a logger would be a more configurable option.  Since the network library should just _work_ I would set the level to be fine or finer or something.

Since it should just work, this is why I have no logging at all in the threadly library.  It's not the job of others to debug my application.